farm_ng/core/CMakeLists.txt

```diff
@@ -5,3 +5,6 @@ core
 add_subdirectory(logging)
 add_subdirectory(enum)
 add_subdirectory(misc)
+if(Protobuf_FOUND)
+  add_subdirectory(serial)
```

Comment:
Suggest a different name than serial (as it might conflict with embedded serial UART terminology).  Perhaps proto, protobuf, or serialization.